### DISCUSSION TITLE
Is there a function that can immediately synchronize data to the vue data without waiting for the flush time to update?

### DISCUSSION THREAD

**Discussion by Brandon-Xu** [Kitware/trame#478] [Q&A] [answered]
I have a function, to calculate a object volume by Layer.
but it will take a lot of time as the object I calculated has a lot of cells.

so, I prepared a callback function that can retrieve the current computation progress, 
and I plan to update the Vue data with it immediately each time it's called, so that users can see the current progress of the calculation. 

However, when I try to update the progress using `server.state.dirty('progress_info')`. 
but wslink does not execute immediately, even if I add `server.flush()`. 

It still waits until the `cal_layer_distribution` execution is complete and then pushes all the accumulated calls at once, 
even though I can clearly see from the console that there's a significant interval between each callback.

Is there any way to immediately update the vue data by wslink?

``` python
def callback(progress):
    server.state.progress_info = int(progress * 100)
    server.state.dirty("progress_info")
    server.state.flush()

def cal_layer_distribution(mesh:pv.DataSet, ids:list[int]=None, callback:Callable=None) -> tuple[float, dict]:
    layer_distribution = {}
    total_volume = abs(mesh.volume)
    total_progress = len(ids)

    if "Layer" not in mesh.array_names:
        raise ValueError("Array named 'Layer' is not in the current mesh!")
    for layer_id in ids:
        callback(layer_id/total_progress)
        volume = abs(mesh.threshold((layer_id, layer_id), scalars="Layer").volume)  # The steps that take the most time
        layer_distribution[layer_id] = {"volume": volume, "percentage": volume/total_volume}
    callback(1)

    return (total_volume, layer_distribution)
```

**Comment by jourdain** [comment, marked as answer]
https://github.com/Kitware/trame/discussions/453

**Comment by jourdain** [comment]
In short if you are locking the main thread from asyncio and therefore, you are just stacking the network requests without giving it any room to be managed right away.
